### PR TITLE
0.2.x Backport of stef work to date

### PR DESCRIPTION
**Pull request by maackle**
### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## \[Unreleased\]
 
+- **BREAKING**: gossip protocol has breaking changes:
+  - Initiate and Accept messages send a list of agent hashes rather than full agent info
+
```
Comment (maackle) [inline crates/kitsune_p2p/kitsune_p2p/CHANGELOG.md:12]: TODO, must revert this change for 0.2

**Comment by ThetaSinner** [comment]
Are all these changes on `develop`? It looks like at least some of them aren't to me. Ideally backports would come from cherry-picked commits from develop because we currently aren't reviewing backports

**Comment by maackle** [comment]
@ThetaSinner no, this is all from my still-unmerged work beginning with `stef` and ending (so far) with `stef-fetchpool-2`. I wanted to begin to get the backport ready now as a sanity check, because as part of the backport I need to revert a breaking change I made on the 0.3 line.

**Comment by maackle** [comment]
In particular I want to build a binary and test it out on an existing network, to make sure the gossip changes are compatible, and to test the new action recording ability

**Comment by maackle** [comment]
We're not backporting this
